pkg/blockstore/shreds.go

```diff
@@ -83,12 +90,11 @@ func DataShredsToEntries(meta *SlotMeta, shreds []shred.Shred) (entries []Entrie
 		if len(entryBytes) == 0 {
 			continue
 		}
-		dec := bin.NewBinDecoder(entryBytes)
-		var subEntries struct {
-			NumEntries uint64 `bin:"sizeof=Entries"`
-			Entries    []shred.Entry
-		}
-		if err := dec.Decode(&subEntries); err != nil {
+		dec := decoderPool.Get().(*bin.Decoder)
+		dec.Reset(entryBytes)
```

Comment:
Can we avoid using a pool and heap alloc here by creating a `bin.Decoder` that lives on the stack?
And have you seen any performance gains using a pool?

Reply:
The use of a pool for decoders slightly decreased the `allocs/op` in the benchmarks.